NDTensors/ext/NDTensorsAMDGPUExt/indexing.jl

```diff
@@ -0,0 +1,20 @@
+using NDTensors.Expose: Exposed, expose, parent, unexpose
+using NDTensors.GPUArraysCoreExtensions: cpu
+function Base.getindex(E::Exposed{<:ROCArray})
+  return AMDGPU.@allowscalar unexpose(E)[]
+end
+
+function Base.setindex!(E::Exposed{<:ROCArray}, x::Number)
+  AMDGPU.@allowscalar unexpose(E)[] = x
```

Comment:
I would prefer to call this from where it is defined, i.e. `GPUArraysCore.@allowscalar`. Probably we should put `using GPUArraysCore: @allowscalar` at the top of the file and then just use `@allowscalar` in the code.

Reply:
Makes sense, I updated that code. thanks!